order-api/src/main/java/com/zerobase/gamesell/order/application/CartApplication.java

```diff
@@ -0,0 +1,149 @@
+package com.zerobase.gamesell.order.application;
+
+import static com.zerobase.gamesell.order.exception.ErrorCode.AGE_RESTRICTION_VIOLATION;
+import static com.zerobase.gamesell.order.exception.ErrorCode.NOT_FOUND_GAME;
+
+import com.user.gamedomain.domain.Dto.UserDto;
+import com.zerobase.gamesell.order.client.UserClient;
+import com.zerobase.gamesell.order.domain.dto.CartResponse;
+import com.zerobase.gamesell.order.domain.game.AddGameCartForm;
+import com.zerobase.gamesell.order.domain.model.Game;
+import com.zerobase.gamesell.order.domain.redis.Cart;
+import com.zerobase.gamesell.order.domain.repository.GameRepository;
+import com.zerobase.gamesell.order.exception.CustomException;
+import com.zerobase.gamesell.order.service.CartService;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CartApplication {
+  private final CartService cartService;
+  private final GameRepository gameRepository;
+  private final UserClient userClient;
+
+  public Cart addCart(Long userId, AddGameCartForm form) {
+
+    Game game = gameRepository.findBySellerIdAndId(
+        form.getSellerId(), form.getId())
+        .orElseThrow(() -> new CustomException(NOT_FOUND_GAME));
+
+    UserDto userDto = userClient.getUser(userId);
+    int userAge = Period.between(userDto.getBirthDate(), LocalDate.now()).getYears();
+
+    if(userAge < game.getContentRating()) {
+      throw new CustomException(AGE_RESTRICTION_VIOLATION);
+    }
```

Comment:
```suggestion
    int userAge = Period.between(userDto.getBirthDate(), LocalDate.now()).getYears();

    if(Period.between(userDto.getBirthDate(), LocalDate.now()).getYears() < game.getContentRating()) {
      throw new CustomException(AGE_RESTRICTION_VIOLATION);
    }
```
userAge 가 두번이상 사용되지 않으므로 굳이 선언하실 필요가 없을 것 같습니다